### PR TITLE
Update FederatedNamespacePlacement to be namespace-scoped

### LINKED ISSUE #107: Namespace placement should be namespaced
Currently namespace placement is cluster-scoped, but this complicates the application of fine-grained authz.  If only the 'tenant' for a given namespace should be allowed to affect its placement, then an explicit rbac rule for just that placement resource would be required.  Changing the namespace placement type to be namespaced would allow for namespace-based authz instead.

### PR DESCRIPTION
**Pull request by marun**
Fixes #107 (and aids #197).

### CODE REVIEW

File: pkg/apis/core/v1alpha1/federatednamespaceplacement_types.go [hunk 1]
```diff
@@ -32,7 +32,6 @@ type FederatedNamespacePlacementStatus struct {
 
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-// +genclient:nonNamespaced
```
[the remark this answers is not in the document]
Comment (shashidharatd) [reply pkg/apis/core/v1alpha1/federatednamespaceplacement_types.go]: > One question is how to make the ns placement policy as namespace scoped?

@gyliu513, this pr makes the FederatedNamespacePlacement namespace scoped. i could not understand your question. Did you mean something different?

> Does the example https://github.com/kubernetes-sigs/federation-v2/blob/master/example/sample1/federatednamespace-placement.yaml need some update as well? Also I found that there is no test file for federatednamespaceplacement_types.go, do you know why?

Thanks for your careful observation @gyliu513, Although not critical, these were missed.

[the remark this answers is not in the document]
Comment (marun) [reply pkg/apis/core/v1alpha1/federatednamespaceplacement_types.go]: I don't think there is much value in the generated unit tests.  I also don't think the examples are worth maintaining due to the nature of the fed primitives (and the fact they are intended to be generated dynamically at some point).

[the remark this answers is not in the document]
Comment (gyliu513) [reply pkg/apis/core/v1alpha1/federatednamespaceplacement_types.go]: > @gyliu513, this pr makes the FederatedNamespacePlacement namespace scoped. i could not understand your question. Did you mean something different?

@shashidharatd @marun I just do not know how to make the namespace placement policy namespace scoped. As far as I know, the namespace placement should be cluster scoped. Can you please share a more concrete use case for this?

@marun Yes, +1 to take the test as low priority. Just do not know why this file is missed.

[the remark this answers is not in the document]
Comment (marun) [reply pkg/apis/core/v1alpha1/federatednamespaceplacement_types.go]: @gyliu513 As per the linked issue, namespace placement should be namespace scoped to minimize the overhead of maintaining authz separation in a multi-tenant deployment.

[the remark this answers is not in the document]
Comment (marun) [reply pkg/apis/core/v1alpha1/federatednamespaceplacement_types.go]: @gyliu513 Regarding the missing test, there was lots of manual fixup in the transition from apiserver-builder to kubebuilder.

**Comment by marun** [comment]
Rebased.

**Pull request merged**